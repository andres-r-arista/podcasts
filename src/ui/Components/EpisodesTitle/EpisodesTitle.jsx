import React from "react";

import PropTypes from "prop-types";

export const EpisodesTitle = ({ amount }) => {
  return (
    <h1 className="border-b md:border md:m-4 md:shadow-md font-bold truncate max-w-full text-lg h-12 px-2 pt-2" data-testid="episodes-title-amount">{`Episodes: ${amount}`}</h1>
  );
};

EpisodesTitle.propTypes = {
  amount: PropTypes.number.isRequired,
};