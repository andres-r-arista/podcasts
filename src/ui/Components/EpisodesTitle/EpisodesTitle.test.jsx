import { render, screen } from "@testing-library/react";

import { EpisodesTitle } from "./EpisodesTitle";


describe("Tests in <EpisodesTitle />", () => {
  test("must match the snapshot", () => {
    const { container } = render(<EpisodesTitle amount={10} />);
    expect(container).toMatchSnapshot();
  });

  test("must display the amount", () => {
    render(<EpisodesTitle amount={10} />);
    const screenDescription = screen.getByTestId("episodes-title-amount");
    expect(screenDescription).toHaveTextContent("10");
  });

});
