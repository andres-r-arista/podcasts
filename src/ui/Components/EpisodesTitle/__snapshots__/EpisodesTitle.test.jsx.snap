// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <EpisodesTitle /> must match the snapshot 1`] = `
<div>
  <h1
    class="border-b md:border md:m-4 md:shadow-md font-bold truncate max-w-full text-lg h-12 px-2 pt-2"
    data-testid="episodes-title-amount"
  >
    Episodes: 10
  </h1>
</div>
`;
